### ISSUE TITLE
Nested grids drag drop doesn't dispose when hovering over nested grid

### ISSUE THREAD

**Issue by rogerfar** [react-grid-layout/react-grid-layout#1485]
I couldn't reproduce this on codesandbox because dragging doesn't seem to work there, so I made a case on stackblitz:
https://stackblitz.com/edit/react-qeuac3?file=src%2FApp.js

When you drag the element into the first grid on the right side and the drag it to the left between 0 and 1 you can see that the placeholder doesn't disappear.

It will also trigger 2 drop events, which can be fixed by adding `event.stopPropagation();`, but this causes the ghost overlay to remain and only to disappear when starting another drag.

**Comment by rogerfar** [comment]
I added 
`e.preventDefault();
e.stopPropagation();`

To onDragLeave, onDragEnter and onDrop and that seems to fix it.

@STRML do you see any issues with that, I can create a small PR.

**Comment by thomaspessato** [comment]
hey @rogerfar, do you happen to have that fixed example?

**Comment by rogerfar** [comment]
I can have a go at trying to get the patch in, it's not that easy as the current NPM package is a bit out of date from master.

**Comment by rogerfar** [comment]
@thomaspessato I updated my example here: https://stackblitz.com/edit/react-qeuac3?file=src%2Freact-grid-layout.js.

The only thing I did here was take the current master, build it and pasted the minified output in `react-grid-layout.js`. Then I added `e.stopPropogation`:

```
F(X(t), 'onDragLeave', function(e) {
                  e.preventDefault(),
                    e.stopPropagation(),
                    t.dragEnterCounter--,
                    0 === t.dragEnterCounter && t.removeDroppingPlaceholder();
                }),
                F(X(t), 'onDragEnter', function(e) {
                  e.preventDefault(), e.stopPropagation(), t.dragEnterCounter++;
                }),
                F(X(t), 'onDrop', function(e) {
                  e.preventDefault();
                  e.stopPropagation();
                  var r = t.props.droppingItem,
                    n = t.state.layout,
                    o = n.find(function(t) {
                      return t.i === r.i;
                    });
                  (t.dragEnterCounter = 0),
                    t.removeDroppingPlaceholder(),
                    t.props.onDrop(n, o, e);
                }),
```

**Comment by rogerfar** [comment]
Can this be re-opened? This bug is pretty annoying and I added a PR for it.

**Comment by willh33** [comment]
I need this as well

**Comment by willh33** [comment]
@rogerfar Does drag and drop work with nested grids?

**Comment by rogerfar** [comment]
It does work, it can be a bit wonky with stuff trying to compact that's in the parent or further down the children. There are some bugs, like the PR I put in.

In the end we forked the library and made some big overhauls, ported everything to typescript and replaced the pixel width, height system with CSS Grids to support auto heights.

**Comment by willh33** [comment]
I'll probably have to do the same. For some reason I'm having an issue where trying to drag elements in a sub-grid also drag the sub-grid. Would the onDragStart also need event.stopPropagations() so it doesn't bubble up to the sub-grid?

**Comment by STRML** [comment]
Fixed in #1494
